ifbcat_api/views.py

```diff
@@ -330,36 +330,40 @@ class EventViewSet(PermissionInClassModelViewSet, viewsets.ModelViewSet):
         '-start_date',
     ]
 
-    queryset = models.Event.objects.annotate(
-        realisation_status=Case(
-            When(Q(start_date__gt=timezone.now()), then=Value('future')),
-            When(
-                Q(start_date__lt=timezone.now()) & (Q(end_date__isnull=True) | Q(end_date__lt=timezone.now())),
-                then=Value('past'),
-            ),
-            default=Value('ongoing'),
-            output_field=CharField(),
-        )
-    ).annotate(
-        registration_status=Case(
-            When(
-                Q(registration_opening__gt=timezone.now()),
-                then=Value('future'),
-            ),
-            When(
-                (Q(registration_opening__isnull=False) | Q(registration_closing__isnull=False))
-                & (Q(registration_opening__isnull=True) | Q(registration_opening__lt=timezone.now()))
-                & (Q(registration_closing__isnull=True) | Q(registration_closing__gt=timezone.now())),
-                then=Value('open'),
-            ),
-            When(
-                Q(registration_opening__isnull=True) & Q(registration_closing__isnull=True),
-                then=Value('unknown'),
-            ),
-            default=Value('closed'),
-            output_field=CharField(),
+    queryset = (
+        models.Event.objects.annotate(
+            realisation_status=Case(
+                When(Q(start_date__gt=timezone.now()), then=Value('future')),
+                When(
+                    Q(start_date__lt=timezone.now()) & (Q(end_date__isnull=True) | Q(end_date__lt=timezone.now())),
+                    then=Value('past'),
+                ),
+                default=Value('ongoing'),
+                output_field=CharField(),
+            )
+        ).annotate(
+            registration_status=Case(
+                When(
+                    Q(registration_opening__gt=timezone.now()),
+                    then=Value('future'),
+                ),
+                When(
+                    (Q(registration_opening__isnull=False) | Q(registration_closing__isnull=False))
+                    & (Q(registration_opening__isnull=True) | Q(registration_opening__lt=timezone.now()))
+                    & (Q(registration_closing__isnull=True) | Q(registration_closing__gt=timezone.now())),
+                    then=Value('open'),
+                ),
+                When(
+                    Q(registration_opening__isnull=True) & Q(registration_closing__isnull=True),
+                    then=Value('unknown'),
+                ),
+                default=Value('closed'),
+                output_field=CharField(),
+            )
         )
+        & models.Event.objects.filter(Q(is_draft=False) | Q(is_draft=None))
```

Comment:
as is_draft will be required, this filter will have to be updated

Reply:
May be!
But I think the result will still be the same. Unless there's a performance issue?

Reply:
It's not a matter of performance, juste that if is_draft is required, then testing if it is None is just dead code